pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/ZKMetadataUtils.java

```diff
@@ -69,6 +100,13 @@ private static void updateSegmentZKMetadata(String tableNameWithType, SegmentZKM
       segmentZKMetadata.setStartTime(segmentMetadata.getStartTime());
       segmentZKMetadata.setEndTime(segmentMetadata.getEndTime());
       segmentZKMetadata.setTimeUnit(segmentMetadata.getTimeUnit());
```

Comment:
We can change this to always put `MILLISECONDS ` as the time unit (convert start/end time using the time unit), then we no longer need to worry about time unit when updating the time interval

Reply:
Makes sense. Closing the PR #9472. I haven't removed the logic to convert timestamp using timeUnit since I am not sure about backward compatibility